javascript/ql/lib/semmle/javascript/dataflow/internal/StepSummary.qll

```diff
@@ -69,6 +71,26 @@ private module Cached {
     AccessPath::isAssignedInUniqueFile(global)
   }
 
+  bindingset[fun]
+  pragma[inline_late]
+  private DataFlow::PropRead getStoredPropRead(DataFlow::FunctionNode fun, string storeProp) {
+    result = fun.getAReturn().getALocalSource().getAPropertySource(storeProp)
+  }
+
+  /**
+   * Holds if `loadProp` of `param` is stored in the `storeProp` property of the return value of `fun`.
+   */
+  pragma[nomagic]
+  private predicate summarizedLoadStoreStep(
+    DataFlow::ParameterNode param, DataFlow::FunctionNode fun, string loadProp, string storeProp
+  ) {
+    exists(DataFlow::PropRead read |
+      read = getStoredPropRead(fun, storeProp) and
+      read.getBase().getALocalSource() = param and
+      read.getPropertyName() = loadProp
+    )
```

Comment:
I'm wondering if it should be implemented like this?

```suggestion
    param.getAPropertyRead(loadProp).flowsTo(fun.getAReturn().getALocalSource().getAPropertySource(storeProp))
```

Otherwise we would not be able to handle the variant below with a middle node, right? (but maybe that's intentional?)

```js
function loadStore_mid(x) {
    const mid = x.loadProp
    return { storeProp: mid };
}

Reply:
It works out because `getAPropertySource()` includes a `getALocalSource()` step. I've added the above as a test case to confirm.

Reply:
ah yes, so it does. Thanks for adding the test :+1: